espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/handlers/MobileBackdoor.java

```diff
@@ -0,0 +1,98 @@
+package io.appium.espressoserver.lib.handlers;
+
+import android.app.Activity;
+import android.util.ArrayMap;
+
+import java.lang.reflect.Field;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import io.appium.espressoserver.lib.handlers.exceptions.AppiumException;
+import io.appium.espressoserver.lib.helpers.InvocationOperation;
+import io.appium.espressoserver.lib.http.response.AppiumResponse;
+import io.appium.espressoserver.lib.model.AppiumStatus;
+import io.appium.espressoserver.lib.model.MobileBackdoorMethod;
+import io.appium.espressoserver.lib.model.MobileBackdoorParams;
+
+import static io.appium.espressoserver.lib.helpers.AndroidLogger.logger;
+
+public class MobileBackdoor implements RequestHandler<MobileBackdoorParams, String> {
+
+    @Override
+    public String handle(final MobileBackdoorParams params) throws AppiumException {
+        logger.info("Invoking Backdoor");
+        Activity activity = getActivity();
+
+        List<InvocationOperation> ops = getBackdoorOperations(params);
+
+        Object invocationResult = null;
+
+        Object InvokeOn = activity.getApplication();
+        for (InvocationOperation op : ops) {
+            try {
+                invocationResult = op.apply(InvokeOn);
+                InvokeOn = invocationResult;
+            } catch (Exception e) {
+                e.printStackTrace();
+            }
+        }
+
+        if (invocationResult instanceof Map && ((Map) invocationResult).containsKey("error")) {
+            InvokeOn = activity;
+            for (InvocationOperation op : ops) {
+                try {
+                    invocationResult = op.apply(InvokeOn);
+                    InvokeOn = invocationResult;
+                } catch (Exception e) {
+                    e.printStackTrace();
+                }
+            }
+        }
+
+        return invocationResult.toString();
+    }
+
+    private List<InvocationOperation> getBackdoorOperations(MobileBackdoorParams params) {
+        List<InvocationOperation> ops = new ArrayList<>();
+        List<MobileBackdoorMethod> mobileBackdoorMethods = params.getOpts();
+
+        for (MobileBackdoorMethod mobileBackdoorMethod : mobileBackdoorMethods) {
+            String methodName = mobileBackdoorMethod.getName();
+            List<Object> arguments = new ArrayList<Object>();
```

Comment:
what is the reason to have an intermediate `arguments` array? We could do null->empty list replacement directly in getArgs getter

Reply:
Fixed